go-controller/pkg/node/gateway_init.go

```diff
@@ -208,50 +206,18 @@ func (n *OvnNode) initGateway(subnets []*net.IPNet, nodeAnnotator kube.Annotator
 	gw.loadBalancerHealthChecker = loadBalancerHealthChecker
 	gw.portClaimWatcher = portClaimWatcher
 
+	initGw := func() error {
```

Comment:
@dave-tucker this is fine, though you could just make the postFn signature take a watch factory if you want. Managment port just wouldn't use it.